### PR TITLE
Feat(28):Proposal creation - Tailwind refactor variant

### LINKED ISSUE #28: Proposal creation flow
I am creating just one issue for the whole proposal creation flow even though it has lots of subcomponents, because Tomasz already did majority of the work and from my understanding it just needs to be hooked up to smart contracts to submit the proposal.

To start creating a proposal, user needs to connect their wallet and have 2500 veMENTO in their account. IF they don't, we show the locking component so they can lock.

Then there are two steps in proposal creation flow:
1. Adding title and decription.
2. Adding execution code in json format.
It is important to cache changes user makes in these steps locally, so they don't lose their progress if they close the window for whatever reason.

Then user can review their input and trigger submitting proposal onchain if they are happy with the result, or can go back to make changes.

Once the proposal is successfully created, user should be redirected to the proposal page.

[Component design in Figma](https://www.figma.com/file/QSQsvksBOspBTONiBBFpqa/MENTO-Airgrab?type=design&node-id=4347%3A2&mode=design&t=oeezJpExpJOH1FAN-1)

### PR DESCRIPTION
**Pull request by RyRy79261**
## Overview

This PR leverages the #101 PR & builds off of the existing proposal creation PR to wire up the appropriate write & read actions

## Reviewing instructions

This PR improves some logic around how the data is fetched, synced & updated.

It allows for caching of form input data until the proposal succeeds. 

I refactored the `SeeMore` component to be more generic and have smooth animations.

There are two deviations from the design, the addition of the example code modal, as well as changing the transaction information screen at the end to a modal. I wrapped it in a modal for two reasons, it blocks any actions during the signing process, I felt its more of an informational screen than anything, for an action the user cant stop, so having it more as an alert felt more intuitive, especially in the case that it fails and you go back more more swiftly. This can be undone simply by rending the contents as a screen instead of in the modal.

## Related issues

- closes #55 
- closes #28

### CODE REVIEW

**Review by bayological — APPROVED** [review]
Nice work. I went through the flow and all works as expected. Some small UI issues to be addressed but so far LGTM.

File: src/components/create-proposal/create-proposal-provider.tsx [hunk 1]
```diff
@@ -69,14 +131,107 @@ export const CreateProposalProvider = ({
...
+    submitProposal();
+  }, [resetCreateProposalHook, submitProposal]);
+
+  useEffect(() => {
+    if (creationState === "ready") {
+      setCreationState("mounting");
+    }
+    // Only update on chainID change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [chainId]);
```
Comment (bowd) [inline src/components/create-proposal/create-proposal-provider.tsx:234]: nit: you can probably do this to get rid of the eslint rule:
```ts
useEffect(() => {
    setCreationState((creationState) => (
       creationState === "ready" ? "mounting" : creationState
    ))
  }, [chainId, setCreationState]);
```

File: src/lib/contracts/governor/useProposals.ts [hunk 2]
```diff
@@ -66,8 +71,27 @@ const useProposals = () => {
...
+    [proposals],
+  );
+
+  // useEffect(() => {
+  //   if (blockNumber && blockNumber % 4n === 0n) {
+  //     refetch();
+  //   }
+  //   // Only needed on block changes
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [blockNumber]);
```
Comment (bowd) [inline src/lib/contracts/governor/useProposals.ts:89]: remove?

File: src/styles/brand.scss [hunk 3]
```diff
@@ -257,6 +257,8 @@ $themes: (
 }
 
 @mixin default-input-error-message {
+  // "p2-1 text-sm font-semibold text-error-dark"
+
```
Comment (bowd) [inline src/styles/brand.scss:261]: remove?

**Review by bowd — APPROVED** [review]
Not an in-depth review but left a few small comments.

File: src/components/_shared/tabs/tab-list.component.tsx [hunk 4]
```diff
@@ -41,6 +41,7 @@ export const TabList = ({
           <button
             key={index}
             className={cn(
+              // TODO: CVA
```
[the remark this answers is not in the document]
Comment (RyRy79261) [reply src/components/_shared/tabs/tab-list.component.tsx:44]: Nah its meant for #107

File: src/components/create-proposal/create-proposal-provider.tsx (hunk 1 above)
Comment (RyRy79261) [reply src/components/create-proposal/create-proposal-provider.tsx:234]: @bowd  That's not what this error is blocking, the issue is it wants the `creationState` variable as well, which will trigger a change endlessly

image

**Pull request merged**